### PR TITLE
Default to `ast:false` and do less work when loading core

### PR DESCRIPTION
**Pull request by loganfsmyth**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  | Y, `ast` now defaults to `false`
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-core/src/config/index.js [hunk 1]
```diff
@@ -3,7 +3,6 @@
 import path from "path";
 import * as context from "../index";
 import Plugin from "./plugin";
-import merge from "lodash/merge";
```
Comment (loganfsmyth) [inline packages/babel-core/src/config/index.js]: Once we make dependencies lazy-load, not loading lodash during anywhere in `src/config` makes a big difference.

File: packages/babel-core/src/transformation/index.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ export function runSync(
   return {
     metadata: file.metadata,
     options: opts,
-    ast: opts.ast !== false ? file.ast : null,
```
Comment (loganfsmyth) [inline packages/babel-core/src/transformation/index.js]: Most callers of Babel don't use this and it is a huge structure. Not including it in the output means when we land caching logic in core one day, we'll have less data to cache by default.

File: packages/babel-helpers/src/helpers.js [hunk 3]
```diff
@@ -23,9 +19,9 @@ helpers.typeof = defineHelper(`
 
     return _typeof(obj);
   }
-`);
+`;
 
-helpers.jsx = defineHelper(`
+helpers.jsx = () => template.program.ast`
```
Comment (loganfsmyth) [inline packages/babel-helpers/src/helpers.js]: This just avoids parsing the helper snippets until the first time they are used.

File: packages/babel-core/src/config/index.js [hunk 4]
```diff
@@ -161,6 +160,30 @@ export default function loadConfig(inputOpts: mixed): ResolvedConfig | null {
...
+      const targetObj = (target.generatorOpts = target.generatorOpts || {});
+      mergeObject(targetObj, generatorOpts);
+    } else {
+      const val = source[k];
+      if (val !== undefined) target[k] = (val: any);
+    }
+  }
+}
+
+function mergeObject<T: {}>(target: T, source: T) {
```
Comment (Andarist) [inline packages/babel-core/src/config/index.js]: maybe we could figure out some better name for this one? i had to look into the implementation to see how it is different from `Object.assign`, maybe just smth like `safeMerge`?

File: packages/babel-helpers/src/helpers.js [hunk 5]
```diff
@@ -5,11 +5,7 @@ import template from "@babel/template";
 const helpers = {};
 export default helpers;
 
-// Helpers never include placeholders, so we disable placeholder pattern
-// matching to allow us to use pattern-like variable names.
-const defineHelper = template.program({ placeholderPattern: false });
-
-helpers.typeof = defineHelper(`
+helpers.typeof = () => template.program.ast`
```
Comment (Andarist) [inline packages/babel-helpers/src/helpers.js]: was removing `placeholderPattern` for those helper templates deliberate?

Comment (loganfsmyth) [reply packages/babel-helpers/src/helpers.js]: Yeah, the `.ast` variant already returns the AST unchanged, so there aren't any placeholders to bother searching for.

File: packages/babel-core/src/config/index.js (hunk 4 above)
Comment (loganfsmyth) [reply packages/babel-core/src/config/index.js]: How about `mergeDefaults` since that seems to be the common name for this operation?

Comment (Andarist) [reply packages/babel-core/src/config/index.js]: Sounds good to me 😃

File: packages/babel-helpers/src/helpers.js (hunk 5 above)
Comment (Andarist) [reply packages/babel-helpers/src/helpers.js]: Cool, good to know. Thanks for clarifying

File: packages/babel-core/src/transformation/index.js (hunk 2 above)
Comment (Sunbridger) [reply packages/babel-core/src/transformation/index.js]: o

**Pull request merged**